src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length < 1) {
+      return '';
+    }
+
+    if (this.length < 2) {
+      return String(this[0]);
+    }
+
+    const gap = (separator === undefined || separator === null)
+      ? `${separator}`
+      : separator;
+
+    let output = (Object.is(this[0], null) || Object.is(this[0], undefined))
+      ? ''
+      : this[0];
+
+    for (let i = 1; i < this.length; i++) {
+      const item = (Object.is(this[i], null) || Object.is(this[i], undefined))
+        ? ''
+        : this[i];
```

Comment:
and also here

Reply:
yes, this check is needed, otherwise the function writes null and undefined as a string item ```'null' & 'undefined'```, instead of ```'<empty value>'``` 
So, I made it more readable and put inside the loop
fixed 👌